mitodl/unified-ecommerce-frontend
# Initial Setup

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/6113

### Description (What does it do?)
Adds scaffolding for the Unified Ecommerce frontend. In particular:
- NextJS-based react site
- A few example example routes, with authentication and an API call for user info
- linting, formatting, tests, plus ci script


### How can this be tested?
1. Follow setup directions in readme.
2. Visit the site in an incognito window:
     - If you visit http://localhost:8072 you should get a helpful error message
     - if you visit the correct site, you should be redirected to keycloak for authentication.
     - After authenticating, your username should appear in header
3. View the homepage and `/cart`

### Review notes
⚠️ The API Client files are auto-generated and copied from the backend repo; after those are published as a package, they will be removed.
